src/ui/map.js

```diff
@@ -1409,6 +1409,25 @@ class Map extends Camera {
         return this._update();
     }
 
+    /**
+     * Removes feature state, setting it back to the default behavior. If only
+     * source is specified, removes all feature states of that source. If
+     * feature.id is also specified, removes all keys for that feature's state.
+     * If key is also specified, removes that key from that feature's state.
+     *
+     * @param {Object} feature Feature identifier. Feature objects returned from
+     * {@link Map#queryRenderedFeatures} or event handlers can be used as feature identifiers.
+     * @param {string | number} feature.id (optional) Unique id of the feature. Optional if key is not specified.
+     * @param {string} feature.source The Id of the vector source or GeoJSON source for the feature.
+     * @param {string} [feature.sourceLayer] (optional)  *For vector tile sources, the sourceLayer is
+     *  required.*
+     * @param {string} key (optional) The key in the feature state to reset.
+    */
+    removeFeatureState(feature: { source: string; sourceLayer?: string; id?: string | number; }, key?: string) {
```

Comment:
Can we rename the "feature" argument to something broader here (and in the matching style.js setFeatureState)? On my first reading I totally missed that you could leave out a feature ID in order to remove an entire layer.

Reply:
now called `target` 👍